core/io/mod.rs

```diff
@@ -145,8 +160,9 @@ pub struct Completion {
 
 #[derive(Debug)]
 struct CompletionInner {
-    pub completion_type: CompletionType,
+    completion_type: CompletionType,
     is_completed: Cell<bool>,
+    error: std::sync::OnceLock<CompletionError>,
 }
```

Comment:
I am wondering if it is possible to hold `Result<T, E>` itself, then you dont have to do checks like these: 

> while !c.has_error() && !c.is_completed() {}

then you wrap it in an `Option`, in cases like these you could check if `Result` is `None` or `Some`